android/app/src/main/java/com/coboltforge/dontmind/multivnc/VncCanvasActivity.java

```diff
@@ -988,6 +995,34 @@ private void sendSpecialKeyAgain() {
 		vncCanvas.sendMetaKey(lastSentKey);
 	}
 
+  private void setZoom() {
+
+    AlertDialog.Builder b = new AlertDialog.Builder(this)
+      .setTitle(getString(R.string.set_zoom_dialog_caption));
+
+    final EditText input = new EditText(this);
+    input.setInputType(InputType.TYPE_CLASS_NUMBER);
+    b.setView(input);
+
+    b.setPositiveButton("OK", new DialogInterface.OnClickListener() { 
+      @Override
+      public void onClick(DialogInterface dialog, int which) {
+        float zoomPercent = Float.parseFloat(input.getText().toString());
+        float zoomFactor = zoomPercent / 100.0f;
+        vncCanvas.scaling.setZoom(VncCanvasActivity.this, zoomFactor);
+      }
+    });
+
+    b.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+      @Override
+      public void onClick(DialogInterface dialog, int which) {
+        dialog.cancel();
+      }
+    });
+
+    b.show();
```

Comment:
the dialog is way too wide for a three-digit number

![signal-attachment-2020-06-03-215827](https://user-images.githubusercontent.com/80450/83683351-8b37b880-a5e5-11ea-989e-4e0c491d1617.jpeg)